mmdet/models/backbones/dla.py

```diff
@@ -0,0 +1,640 @@
+import torch
+from torch import nn
+from mmcv.cnn import (build_norm_layer, build_conv_layer, build_plugin_layer,
+                      constant_init)
+import torch.utils.checkpoint as cp
+from mmcv.runner import load_checkpoint
+from mmdet.utils import get_root_logger
+from torch.nn.modules.batchnorm import _BatchNorm
+from ..builder import BACKBONES
+
+
+class BasicBlock(nn.Module):
+
+    def __init__(self,
+                 inplanes,
+                 planes,
+                 stride=1,
+                 dilation=1,
+                 downsample=None,
+                 style='pytorch',
+                 with_cp=False,
+                 conv_cfg=None,
+                 norm_cfg=dict(type='BN'),
+                 dcn=None,
+                 plugins=None):
+        super(BasicBlock, self).__init__()
+        assert dcn is None, 'Not implemented yet.'
+        assert plugins is None, 'Not implemented yet.'
+
+        self.norm1_name, norm1 = build_norm_layer(norm_cfg, planes, postfix=1)
+        self.norm2_name, norm2 = build_norm_layer(norm_cfg, planes, postfix=2)
+
+        self.conv1 = build_conv_layer(
+            conv_cfg,
+            inplanes,
+            planes,
+            kernel_size=3,
+            stride=stride,
+            padding=dilation,
+            dilation=dilation,
+            bias=False)
+
+        self.add_module(self.norm1_name, norm1)
+
+        self.relu = nn.ReLU(inplace=True)
+
+        self.conv2 = build_conv_layer(
+            conv_cfg,
+            planes,
+            planes,
+            kernel_size=3,
+            stride=1,
+            padding=dilation,
+            bias=False,
+            dilation=dilation)
+
+        self.add_module(self.norm2_name, norm2)
+
+        self.downsample = downsample
+        self.stride = stride
+        self.dilation = dilation
+        self.with_cp = with_cp
+
+    @property
+    def norm1(self):
+        return getattr(self, self.norm1_name)
+
+    @property
+    def norm2(self):
+        return getattr(self, self.norm2_name)
+
+    def forward(self, x, identity=None):
+
+        def _inner_forward(x, identity):
+
+            if identity is None:
+                identity = x
+
+            out = self.conv1(x)
+            out = self.norm1(out)
+            out = self.relu(out)
+
+            out = self.conv2(out)
+            out = self.norm2(out)
+
+            if self.downsample is not None:
+                identity = self.downsample(x)
+
+            out += identity
+
+            return out
+
+        if self.with_cp and x.requires_grad:
+            out = cp.checkpoint(_inner_forward, x, identity)
+        else:
+            out = _inner_forward(x, identity)
+
+        out = self.relu(out)
+
+        return out
+
+
+class Bottleneck(nn.Module):
```

Comment:
It seems that BottleNeck and BasicBlock are not changed, can we simply use them from those in resnet.py?

Reply:
@ZwwWayne the arguments of forward function is different

Reply:
I have reduced the codes of BasicBlock and Bottleneck of DLANet by inheriting from the BasicBlock and Bottleneck of ResNet